packages/mdsvex/src/index.ts

```diff
@@ -313,7 +313,10 @@ const _compile = (
 		content: source,
 		filename:
 			(opts && opts.filename) ||
```

Comment:
Running `mdsvex.compile("foo", {filename: "bar"});` generates this warning:

```
mdsvex: Received unknown options: filename. Valid options are: remarkPlugins, rehypePlugins, smartypants, extension, extensions, layout, highlight, frontmatter.
```

... and causes no output to be returned. Perhaps there's more to do here? Happy to help flesh this out a bit...

Reply:
I'll take a peek.

Reply:
If you just add `'filename'` to the `known_opts` array defined at 231-240 of this file, this warning should go away. It is just a convenience in case people pass in unknown options but it doesn't take the `compile` usage into account. 

It doesn't matter if the warning isn't perfect really, deleting the key from the object is a little expensive so simply loosening the warnings is preferable. 

When you have made this change, a test asserting against these warnings will also fail, you can just add `filename` to that string and it should pass again (if you run the tests with `pnpm run test` it should show you what needs changing in the diff). The test is on line 730 of `test/it/mdsvex.test.ts` - [here](https://github.com/wlach/MDsveX/blob/handle-no-options-gracefully-in-compile/packages/mdsvex/test/it/mdsvex.test.ts#L730-L771).

There is also a type error when using this without an options argument. We can fix this by making `opts` optional on line 310:

```ts
opts?: MdsvexCompileOptions
```

Reply:
I did this in a followup PR  #187